### PR TITLE
[String] add tests to the Inflector

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | na
| License       | MIT

Adds a bunch of fixtures to the English and French inflector, these come from the Doctrine component.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Aaaaand... :drum: this makes tests fail. Looks like we have a fix pending :) 
Can you please target 5.4 also?

File: src/Symfony/Component/String/Tests/Inflector/EnglishInflectorTest.php [hunk 1]
```diff
@@ -305,6 +305,331 @@ public static function pluralizeProvider()
...
+            ['diagnosis', 'diagnoses'],
+            ['diagnosis_a', 'diagnosis_as'],
+            ['dictionary', 'dictionaries'],
+            ['dive', 'dives'],
+            ['djinn', 'djinn'],
+            ['domino', 'dominoes'],
+            ['education', 'education'],
+            ['eland', 'eland'],
+            ['elk', 'elk'],
+            // ici
```
Comment (smnandre) [inline src/Symfony/Component/String/Tests/Inflector/EnglishInflectorTest.php:382]: :dart: 
```suggestion
```

**Comment by nicolas-grekas** [comment]
@soyuka up to work on this PR? Otherwise it might be better to close :shrug:

**Comment by soyuka** [comment]
Yes but not sure when... I'll reopen on 5.4 if I take this again, if not anyone can continue on top of the work.

**Pull request closed without merging**